heartbeat/azure_fence_lib.py

```diff
@@ -0,0 +1,406 @@
+#!@PYTHON@ -tt
+import logging, re, time
+
+FENCE_SUBNET_NAME = "fence-subnet"
+FENCE_INBOUND_RULE_NAME = "FENCE_DENY_ALL_INBOUND"
+FENCE_INBOUND_RULE_DIRECTION = "Inbound"
+FENCE_OUTBOUND_RULE_NAME = "FENCE_DENY_ALL_OUTBOUND"
+FENCE_OUTBOUND_RULE_DIRECTION = "Outbound"
+VM_STATE_POWER_PREFIX = "PowerState"
+VM_STATE_POWER_DEALLOCATED = "deallocated"
+VM_STATE_POWER_STOPPED = "stopped"
+VM_STATE_POWER_RUNNING = "running"
+VM_STATE_POWER_STARTING = "starting"
+FENCE_STATE_OFF = "off"
+FENCE_STATE_ON = "on"
+FENCE_TAG_SUBNET_ID = "FENCE_TAG_SUBNET_ID"
+FENCE_TAG_IP_TYPE = "FENCE_TAG_IP_TYPE"
+FENCE_TAG_IP = "FENCE_TAG_IP"
+IP_TYPE_DYNAMIC = "Dynamic"
+MAX_RETRY = 10
+RETRY_WAIT = 5
+
+class AzureSubResource:
+    Type = None
+    Name = None
+
+class AzureResource:
+    Id = None
+    SubscriptionId = None
+    ResourceGroupName = None
+    ResourceName = None
+    SubResources = []
+
+class AzureConfiguration:
+    RGName = None
+    VMName = None
+    SubscriptionId = None
+    Cloud = None
+    UseMSI = None
+    Tenantid = None
+    ApplicationId = None
+    ApplicationKey = None
+    Verbose = None
+
+def get_resource_group_from_metadata():
+    import requests
+    try:
+        r = requests.get('http://169.254.169.254/metadata/instance?api-version=2017-08-01', headers = {"Metadata":"true"})
+        return str(r.json()["compute"]["resourceGroupName"])
+    except:
+        logging.warning("Not able to use metadata service. Am I running in Azure?")
+
+    return None
+
+def get_subscription_id_from_metadata():
+    import requests
+    try:
+        r = requests.get('http://169.254.169.254/metadata/instance?api-version=2017-08-01', headers = {"Metadata":"true"})
+        return str(r.json()["compute"]["subscriptionId"])
+    except:
+        logging.warning("Not able to use metadata service. Am I running in Azure?")
+
+    return None
+
+def fail_usage(message):
+    raise ValueError("%s Run with parameter help to get more information" % message)
+
+# https://stackoverflow.com/questions/715417/converting-from-a-string-to-boolean-in-python
+def ocf_is_true(strValue):
+    return strValue and strValue.lower() in ("yes", "true", "1", "YES", "TRUE", "ja", "on", "ON")

```

Comment:
There is no need to include upper-case words in the comparison list since the string is converted to lower-case before checking.